lib/uvm_agents/uvma_axi5/docs/axi5_agent_documentation.md

```diff
@@ -0,0 +1,430 @@
+  | AXI_ATOP_DATA              | The write data must be equal to 1, 2, 4 or 8 bytes for AtomicStore, AtomicLoad,     |
+  |                            | and AtomicSwap operations.                                                          |
+  |                            |                                                                                     |
+  | AXI_ATOP_COMP_DATA         | The write data must be equal to 2, 4, 8, 16, or 32 bytes for AtomicCompare operation|
+  |                            |                                                                                     |
+  | AXI_ATOP_COMP_ALIGN        | The address must be aligned to a single write data value for AtomicCompare          |
+  |                            |                                                                                     |
+  | AXI_ATOP_COMP_BURST        | The burst type of an AtomicCompare transaction must be INCR if AWADDR points to the |
+  |                            | lower half of the transaction, or WRAP if AWADDR points to the upper half of        |
+  |                            | the transaction.                                                                    |
+  |                            |                                                                                     |
+  | AXI_ATOP_ACCESS            | AWLOCK must be 0 for Atomic operation                                               |
+  |                            |                                                                                     |
+  | x_amo_id                   | The ID used to identify an Atomic transactions must not be used for other           |
+  |                            | transactions that are outstanding at the same time.                                 |
+  |                            |                                                                                     |
+  | axi4_errs_amo_rid          | The slave must send a read data that follow the address that it relates to          |
+  |                            |                                                                                     |
+  | amo_r_comp_len             | The length of the read transaction must be the half of the write transaction        |
+  |                            | for AtomicCompare.                                                                  |
+  |                            |                                                                                     |
+  | amo_comp_len               | The length of the read transaction must be equal to the lenght of write transaction |
+  |                            | for Atomic operation except AtomicCompare                                            |
+
+## Directory Structure
+
+-   docs: Documents describing the AXI.
+
+-   examples: Samples for users
+
+-   src: Source code for this package
+
+## Packages dependencies
```

Comment:
There is also a dependency on `read_elf()`, `get_section()` and `read_section_sv()` (although I am not sure we should keep these).

Reply:
I don't think this is necessary because these functions are related to CVA6.